### PR TITLE
Add FallbackResolvers to tailcfg, and plumb through into DNS config.

### PR DESCRIPTION
**Pull request by danderson**

### CODE REVIEW

File: ipn/ipnlocal/local.go [hunk 1]
```diff
@@ -1635,6 +1635,38 @@ func (b *LocalBackend) authReconfig() {
...
+		// Set FallbackResolvers as the default resolvers in the
+		// scenarios that can't handle a purely split-DNS config. See
+		// https://github.com/tailscale/tailscale/issues/1743 for
+		// details.
+		switch {
+		case len(dcfg.DefaultResolvers) != 0:
+			// Default resolvers already set.
+		case len(dcfg.Routes) == 0 && len(dcfg.Hosts) == 0 && len(dcfg.AuthoritativeSuffixes) == 0:
+			// No settings requiring split DNS, no problem.
+		case (version.OS() == "iOS" || version.OS() == "macOS") && !uc.ExitNodeID.IsZero():
```
Comment (bradfitz) [inline ipn/ipnlocal/local.go:1659]: is this right for tailscaled-on-macOS? I don't think so?

File: tailcfg/tailcfg.go [hunk 2]
```diff
@@ -849,6 +849,10 @@ type DNSConfig struct {
 	// Map keys must be fully-qualified DNS name suffixes, with a
 	// trailing dot but no leading dot.
 	Routes map[string][]DNSResolver `json:",omitempty"`
+	// FallbackResolvers is like Resolvers, but is only used if a
+	// split DNS configuration is requested in a configuration that
+	// doesn't work yet without explicit default resolvers.
```
Comment (bradfitz) [inline tailcfg/tailcfg.go:854]: reference #1743 here too?

File: ipn/ipnlocal/local.go [hunk 3]
```diff
@@ -1635,6 +1635,38 @@ func (b *LocalBackend) authReconfig() {
 				set(peer.Name, peer.Addresses)
 			}
 		}
+
+		addFallback := func() {
+			for _, resolver := range nm.DNS.FallbackResolvers {
+				res, err := parseResolver(resolver)
+				if err != nil {
+					b.logf(err.Error())
```
Comment (bradfitz) [inline ipn/ipnlocal/local.go:1643]: the first parameter to logf is a log format, not a string.

so the obvious fix is `logf("%s", err)` but then you get rate-limited on `"%s"` (sigh). So invent some prefix?

**Review by bradfitz — APPROVED** [review]

File: ipn/ipnlocal/local.go (hunk 1 above)
Comment (danderson) [reply ipn/ipnlocal/local.go:1659]: tailscaled-on-macos has no DNS management atm, right?

Do we have a helper to figure out standalone vs. netext/sysext?

**Pull request merged**